prompt_testing/evaluation/reviewer.py

```diff
@@ -0,0 +1,271 @@
+"""
+Human review tools for prompt evaluation.
+"""
+
+import json
+from dataclasses import asdict, dataclass
+from datetime import datetime
+from pathlib import Path
+
+
+@dataclass
+class HumanReview:
+    """Human review of a prompt response."""
+
+    case_id: str
+    prompt_version: str
+    reviewer: str
+    timestamp: str
+
+    # Scores (1-5 scale)
+    accuracy: int  # How accurate is the technical content?
+    clarity: int  # How clear and understandable is it?
+    completeness: int  # Does it cover all important aspects?
+    educational_value: int  # How helpful is it for learning?
+
+    # Qualitative feedback
+    strengths: list[str]
+    weaknesses: list[str]
+    suggestions: list[str]
+    overall_comments: str
+
+    # Comparison (when reviewing multiple versions)
+    compared_to: str | None = None
+    preference: str | None = None  # 'this', 'other', 'similar'
+    preference_reason: str | None = None
+
+
+class ReviewManager:
+    """Manages human reviews and creates comparison interfaces."""
+
+    def __init__(self, results_dir: str):
+        self.results_dir = Path(results_dir)
+        self.reviews_file = self.results_dir / "human_reviews.jsonl"
+
+    def save_review(self, review: HumanReview) -> None:
+        """Save a human review to the reviews file."""
+        self.results_dir.mkdir(parents=True, exist_ok=True)
+
+        with self.reviews_file.open("a") as f:
+            f.write(json.dumps(asdict(review)) + "\n")
+
+    def load_reviews(self, case_id: str | None = None, prompt_version: str | None = None) -> list[HumanReview]:
+        """Load reviews, optionally filtered by case ID or prompt version."""
+        if not self.reviews_file.exists():
+            return []
+
+        reviews = []
+        with self.reviews_file.open() as f:
+            for line in f:
+                review_data = json.loads(line.strip())
+                review = HumanReview(**review_data)
+
+                if case_id and review.case_id != case_id:
+                    continue
+                if prompt_version and review.prompt_version != prompt_version:
+                    continue
+
+                reviews.append(review)
+
+        return reviews
+
+    def create_review_interface(self, case_id: str, responses: dict[str, dict]) -> str:
```

Comment:
We should discuss the utility of this. It's a lot of extra string processing and HTML I inside python.

I'd rather use a real HTML templating system like pug if we go down this route.

Reply:
For future discussion